models/DataObject/ClassDefinition/Helper/OptionsProviderResolver.php

```diff
@@ -34,7 +33,7 @@ public static function resolveProvider(?string $providerClass, int $mode): ?obje
     {
         return self::resolve($providerClass, function ($provider) use ($mode) {
             return ($mode == self::MODE_SELECT && ($provider instanceof SelectOptionsProviderInterface))
-                || ($mode == self::MODE_MULTISELECT && ($provider instanceof MultiSelectOptionsProviderInterface));
+                || ($mode == self::MODE_MULTISELECT && ($provider instanceof SelectOptionsProviderInterface));
```

Comment:
Should we keep the `MultiSelectOptionsProviderInterface` here? It's only deprecated, and people need time to migrate.

Reply:
You are rigtht.

Reply:
Yes, reintroduced the condition.